### PR TITLE
updated windows CI

### PR DESCRIPTION
**Pull request by sezero**
Please also merge https://github.com/fragglet/lhasa/pull/47 which is needed for dll generation.

### CODE REVIEW

File: .github/workflows/make.yml [hunk 1]
```diff
@@ -51,24 +51,32 @@ jobs:
...
+        msystem: ${{ matrix.installs }}
         update: true
-        install: autoconf automake libtool gcc make groff zip dos2unix
+        install: base-devel git autoconf automake libtool groff zip dos2unix ${{ matrix.pkg }}
+    - uses: actions/checkout@v3
     - name: Find Git version
       id: version
       run: |
         export VERSION=$(git rev-parse --short HEAD)
         echo "VERSION=$VERSION-win32" >> $GITHUB_OUTPUT
```
Comment (fragglet) [inline .github/workflows/make.yml:77]: Suffix should be different for 64-bit version?

**Comment by sezero** [comment]
Rebased to master after merge of #47

File: .github/workflows/make.yml [hunk 2]
```diff
@@ -51,24 +51,32 @@ jobs:
...
     steps:
-    - uses: actions/checkout@v3
-    - uses: msys2/setup-msys2@v2
+    - name: Setup msys2
+      uses: msys2/setup-msys2@v2
       with:
+        msystem: ${{ matrix.installs }}
         update: true
-        install: autoconf automake libtool gcc make groff zip dos2unix
+        install: base-devel git autoconf automake libtool groff zip dos2unix ${{ matrix.pkg }}
```
Comment (fragglet) [inline .github/workflows/make.yml:71]: Specifying the target by changing the version of gcc that gets installed seems a bit too magic. Can you use --host instead? eg.
```
. /autogen.sh --host=i686-mingw-w64
```

**Review by fragglet — CHANGES_REQUESTED** [review]

File: .github/workflows/make.yml [hunk 3]
```diff
@@ -51,24 +51,32 @@ jobs:
...
+    - uses: actions/checkout@v3
     - name: Find Git version
       id: version
       run: |
         export VERSION=$(git rev-parse --short HEAD)
         echo "VERSION=$VERSION-win32" >> $GITHUB_OUTPUT
     - name: configure
-      run: ./autogen.sh --disable-shared
+      run: ./autogen.sh
```
[the remark this answers is not in the document]
Comment (sezero) [reply .github/workflows/make.yml:79]: I actually meant it to see dll generation actually happens. But windows package generation and upload needs some updating after that I guess, which don't know anything about.

Should I revert removal of `--disable-shared`

File: .github/workflows/make.yml (hunk 1 above)
Comment (sezero) [reply .github/workflows/make.yml:77]: I don't think so

File: .github/workflows/make.yml (hunk 2 above)
Comment (sezero) [reply .github/workflows/make.yml:71]: Let me see..

File: .github/workflows/make.yml (hunk 1 above)
Comment (sezero) [reply .github/workflows/make.yml:71]: I misunderstood that thing. fixing..

File: .github/workflows/make.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (sezero) [reply .github/workflows/make.yml]: > I actually meant it to see dll generation actually happens. But windows package generation and upload needs some updating after that I guess, which don't know anything about.
> 
> Should I revert removal of `--disable-shared` ?

Should I revert removal of `--disable-shared` ?

File: .github/workflows/make.yml (hunk 2 above)
Comment (sezero) [reply .github/workflows/make.yml]: Done

File: .github/workflows/make.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (sezero) [reply .github/workflows/make.yml]: I went ahead and put back --disable-shared, added TODO notes in there as to why it is needed, i.e. pkg/win32 doesn't actually package the dll and the import lib but only the exe.

**Comment by sezero** [comment]
Squashed all commits. Should be ready.

**Review by fragglet — APPROVED** [review]
I might rework this a bit but I'm fine with merging it as it is.

**Pull request merged**